colorls.rb

```diff
@@ -124,21 +172,28 @@ def options(content)
   end
 end
 
-args = *ARGV
+args         = *ARGV
+report       = false
+sort         = false
+one_per_line = false
 
-if args.include?('--report') || args.include?('-r')
-  report = true
-  args -= %w[--report -r]
-else
-  report = false
+args.each do |arg|
+  report       = true if ['--report', '-r'].include?(arg)
+  one_per_line = true if arg == '-1'
+
+  match = arg.match(/^--sort=?(.*)?$/)
+
+  if match
+    sort = match.captures[0] == 'dirs-first' ? 'dirs-first' : true
+  end
 end
```

Comment:
If you take this suggestion, there won't be any need to iterate through `args` like `args.each do |arg|`. Rather, simply `args.include?` can be used (like previously).